### PR TITLE
feat: CI tests, linting, formatting, typechecking, SDK version bump

### LINKED ISSUE #20: Run tests in CI
This tap has some tests under `tap_postgres/tests` that would be good to run in CI.

### PR DESCRIPTION
**Pull request by qbatten**
Close #20 and #23.

Did a few things here:
- Ran linting, formatting, typechecking on the whole repo & made requisite changes to the code.
- Add CI testing (& linting, formatting, typechecking, etc), based mostly on [tap-stackexchange's setup](https://github.com/MeltanoLabs/tap-stackexchange/tree/main/.github/workflows).
- Bumped version of SDK to 0.19.0 (needed to loosen tox version req in order to make that happen), bump black version.
- Add pre-commit package and config.

All of the code changes are just things that the various checks asked for. There aren't any new, separate code changes here.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    env:
+      PIP_CONSTRAINT: .github/workflows/constraints.txt
+    strategy:
+      matrix:
+        include:
+        - {python-version: "3.10", session: "lint", os: "ubuntu-latest" }
+        - {python-version: "3.10", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.9", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.8", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.7", session: "pytest", os: "ubuntu-latest" }
```
Comment (visch) [inline .github/workflows/test.yml:33]: I think this is going to fail with our `cached_property` use, we could just not support `3.7`

File: .github/workflows/constraints.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pip==23.0
```
Comment (visch) [inline .github/workflows/constraints.txt:1]: Cool, this should be covered by dependabot right now too. So this should start getting auto prs to get updated as well.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      run: |
+        pip install pip
+        pip --version
+    - name: Install Poetry
+      run: |
+        pipx install poetry
+        poetry --version
+    - name: Install tox
+      run: |
+        pipx install tox
```
Comment (visch) [inline .github/workflows/test.yml:55]: I normally just go `poetry run tox -e lint` but this works too. 

I think I like `poetry` more as I keeps the version stuck to `pyproject.yaml` and will remove the `tox` need in constaints.txt. Can swap to that? 

Take a look at `target-postgres` https://github.com/MeltanoLabs/target-postgres/blob/main/.github/workflows/ci_workflow.yml

File: .precommit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.3.0
+  hooks:
+  - id: check-yaml
+  - id: end-of-file-fixer
+  - id: trailing-whitespace
+- repo: https://github.com/psf/black
+  rev: 22.10.0
+  hooks:
+  - id: black
```
Comment (visch) [inline .precommit-config.yaml:11]: Add MyPy (just adding for an issue to get created)

File: tox.ini [hunk 5]
```diff
@@ -27,6 +28,7 @@ commands =
 [testenv:format]
 # Attempt to auto-resolve lint errors before they are raised.
 # To execute, run `tox -e format`
+allowlist_externals = poetry
```
Comment (visch) [inline tox.ini:31]: Doesn't this cascade from line 9? I'm not even sure we really need this

File: tox.ini [hunk 6]
```diff
@@ -35,6 +37,7 @@ commands =
 [testenv:lint]
 # Raise an error if lint and style standards are not met.
 # To execute, run `tox -e lint`
+allowlist_externals = poetry
```
Comment (visch) [inline tox.ini:40]: ditto above

**Review by visch — CHANGES_REQUESTED** [review]
Thanks for cleaning up some of this stuff! Really appreciated

File: .github/workflows/test.yml (hunk 1 above)
Comment (qbatten) [reply .github/workflows/test.yml:33]: It passes

File: tox.ini (hunk 5 above)
Comment (qbatten) [reply tox.ini:31]: Yeah actually the real change that was needed is whitelist_externals -> allowlist_externals, this stuff is cruft from my troubleshooting that. Removing

File: .github/workflows/test.yml (hunk 3 above)
Comment (qbatten) [reply .github/workflows/test.yml]: I think I resolved this the way you wanted me to but lmk if I misunderstood or made a mistake when doing that switch.

**Review by visch — APPROVED** [review]
Looks good to me. Tests should pass! 🤞

**Pull request merged**